docs/Contributing/windows-mdm-glossary-and-protocol.md

```diff
@@ -57,6 +57,84 @@ The certificate created through the WSTEP process is used to authenticate mTLS b
+    |--------------|-------|--------------------------------------------------------------------------------------------------------------------|
+    | unknown      | 0     | Device enrollment state is unknown                                                                                 |
+    | enrolled     | 1     | Device is Enrolled.                                                                                                |
+    | pendingReset | 2     | Enrolled but it's enrolled via enrollment profile and the enrolled profile is different from the assigned profile. |
+    | failed       | 3     | Not enrolled and there is enrollment failure record.                                                               |
+    | notContacted | 4     | Device is imported but not enrolled.                                                                               |
+    | blocked      | 5     | Device is enrolled as userless, but is blocked from moving to user enrollment because the app failed to install.   |
+
+  - `EnrollmentType`
+    According to [this PDF](https://winprotocoldoc.blob.core.windows.net/productionwindowsarchives/MS-MDE2/%5BMS-MDE2%5D.pdf) it can have three different values.
+
+    Device, Full, and AppManaged
+
+    From what I've seen, value 6 on AAD, 1 on manual
+  - `isFederated`
+    According to [this web page](https://learn.microsoft.com/en-us/windows/client-management/federated-authentication-device-enrollment), being federated means that the MDM
+    endpoints and details were fetched from a Discovery endpoint,
+    instead of being manually installed. The page does not make mention
+    of the specific registry key, but we are making an assumption that
+    it means the same thing.
+  - `ProviderID`
+    Set during enrollment. In our case it's the word "Fleet".
+  - `RenewalPeriod`
+    Set during enrollment. Period to renew WSTEP certificate.
+  - `RenewErrorCode`
+    Presumably set if there is an error renewing WSTEP certificate.
+  - `RenewROBOSupport`
+    According to [this post](https://call4cloud.nl/2022/10/fullmetal-certificate-the-revenge-of-renewal/) this means "Request On Behalf Of". 
+    It seems to have to do with automatic certificate renewal
+  - `RenewStatus`
+    Status of the renewal
+  - `RenewTimestamp`
+    Presumably the timestamp of the last renewal
+  - `RootCertThumbPrint`
+    The thumbprint of the WSTEP root certificate
+  - `SID`
+    Security Identifier
+  - `UPN`
+    User Principal Name of the user that enrolled the device
+  - `AADResourceID`
+    Appears to be the domain of the server managing the enrollment,
+    always appears to be present on machines enrolled through Microsoft
+    Entra (Azure Active Directory)
+  - `AADTenantID`
+    Also related to Azure Active Directory, and also appears to be
+    present at the same time as AADResourceID.
+- `HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Provisioning\Diagnostics\AutoPilot`
+  Autopilot provisioning diagnostic data
```

Comment:
nice! thanks for this! you might want to send it in a separate PR to decouple the docs review from the bugfix. Up to you!

Reply:
Sure! I can do that

Reply:
On second thought I'm going to leave it for this branch, I would have to destroy it and re-create it to. I'll put them into separate PRs in the future 😅

Reply:
if helps, I have done this in the past, and Rachael's review is not longer required without having to create a new PR, but absolutely up to you, not required by any means